library/library_info.ml

```diff
@@ -0,0 +1,32 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+(*i*)
+open Names
+(*i*)
+
+type t = Comment of string | Deprecation of Deprecation.t
+
+let info_cat = CWarnings.create_category ~name:"info" ()
+
+let warn_library_info =
+  CWarnings.create ~category:info_cat ~name:("info-library")
+  Pp.(fun (dp,s) ->
+    str "Library " ++ DirPath.print dp ++ str ":" ++ spc () ++ str s)
+
```

Comment:
This doesn't belong to the current PR but should come after https://github.com/coq/coq/pull/18248

Reply:
I don't understand why you say this doesn't belong to the current PR.

Reply:
Do we agree that, if not the implementation, at least the user facing syntax for triggering other warnings than deprecations is shared with https://github.com/coq/coq/pull/18248 ?
Then that syntax should be discussed in a single place, either https://github.com/coq/coq/pull/18248 or here. And since it's the whole purpose of https://github.com/coq/coq/pull/18248 it seems more natural to have that discussion there. Then the current PR could either:
1. depend on https://github.com/coq/coq/pull/18248 (but that should be explicitly mentionned in the top message) and wait for it
2. or not have anything to do with that feature, probably be merged before https://github.com/coq/coq/pull/18248 and then https://github.com/coq/coq/pull/18248 will be rebased on top of it (or a third PR will come with the above code)

Considering the discussion in https://github.com/coq/coq/pull/18248 is seemingly not making progress whereas the current PR seems almost ready in case 2, I have a strong preference for 2.

Reply:
> at least the user facing syntax for triggering other warnings than deprecations is shared with #18248?

Yes

> And since it's the whole purpose https://github.com/coq/coq/pull/18248 it seems more natural to have that discussion there.

Either place is ok to me.

Both 1. and 2. are "in principle" ok to me, but I'd like the current PR #18193 (and its `information` attribute) to be used for #18032, so my only preference is that we find a way to converge on the syntax needed for #18032 before 8.19. So, unless I misunderstood your point, I arrive to the same conclusion as you, namely to solution 2.

If ever the attribute syntax `information` needs time beyond the 8.19 deadline, a solution would be to use it only internally (in Vector.v) but without officially documenting it yet.